test/API/H5_api_test.c

```diff
@@ -147,10 +168,75 @@ usage(FILE *stream)
     fprintf(stream, "async       run only the async interface tests\n");
 }
 
+static herr_t
+H5_api_test_setup_container_names(const char *prefix, const char *filename)
```

Comment:
Simplified the logic here a bit by creating MT filenames up front and re-using them when creating/destroying the containers.